backend/server.go

```diff
@@ -152,17 +151,17 @@ var roleCast = map[string]gamelogic.Role{
 }
 
 func AssignRolePool(game *gamelogic.Game, responseWriter http.ResponseWriter, request *http.Request) (error, int) {
-	err, rolesString := getFirstOccuranceInUrlParam(request, "roles")
-	if err != nil {
-		return err, http.StatusBadRequest
+	rolesReq := struct{ Roles []string }{}
+
+	if err := json.NewDecoder(request.Body).Decode(&rolesReq); err != nil {
+		return errors.New("Invalid Json"), http.StatusBadRequest
```

Comment:
I don't know what's go standard here but i define a bunch of go errors closet to the top of this file. Since you don't add any occurrence specific information in might as well make it a single use var